src/pykiso/test_coordinator/test_case.py

```diff
@@ -329,6 +337,27 @@ def func_wrapper(self: BasicTest) -> None:
                             f">>>>>>>>>> Test {retry_nb}/{max_try} failed <<<<<<<<<<"
                         )
                         raise e
+                    elif (
+                        getattr(self, "step_report", False) and self.step_report.header
```

Comment:
You are adding information to the step report so I would rather create a function at step report module level.
You are increasing the complexity of this decorator, so separate it will be a little bit more cleaner.

Reply:
Added a function in step report module.